piksi_tools/fileio.py

```diff
@@ -457,13 +468,23 @@ def main():
             f = FileIO(link)
             try:
                 if args.write:
-                    f.write(args.write[0], open(args.write[0]).read())
+                    f.write(args.write[1], open(args.write[0]).read())
                 elif args.read:
+                    assert len(args.read) <= 2, "Only one source and one destination accepted for sbp fileio read."
+
                     data = f.read(args.read[0])
                     if args.hex:
-                        print(hexdump(data))
+                        if len(args.read) == 2:
+                            with open(args.read[1], 'w') as fd:
```

Comment:
You can invert this if to avoid some duplicate code (also probably want to open the file with `'wb'`):

```python
if len(args.read) == 2:
    with open(args.read[1], 'wb') as fd:
      fd.write(hexdump(data) if args.hex else data)
else:
  print(hexdump(data) if args.hex else data)
```

Reply:
Actually I need to open as 'w' since the hex dump is actually formatted text in this case.  I will take the cleaner flow control though thanks!

Reply:
I take that back, I need to open as binary.